storey/steps/sample.py

```diff
@@ -22,13 +23,22 @@ class SampleWindow(Flow):
     Available options:
         2.1) True - The last event seen will be emitted downstream.
         2.2) False - The last event seen will NOT be emitted downstream.
+    :param count_per_key: Whether to sample up to window size on per key basis. False samples on all events.
+    :param key: The key by which events are sampled. By default (None), events are not sampled by key.
+        Other options may be:
+        Set a '$key' to sample events by the Event.key property.
+        set a 'str' key to sample events by Event.body[str].
+        set a Callable[Any, Any] to sample events by a custom key extractor.
```

Comment:
This line says `Callable[Any, Any]`, the type annotations say `Callable[[Event], str]`.